web/src/beta/features/Editor/tabs/map/LeftPanel/Layers/index.tsx

```diff
@@ -57,30 +63,74 @@ const Layers: React.FC<LayersProps> = ({
       {layers.map(layer => (
         <LayerItem
           key={layer.id}
+          id={layer.id}
           layerTitle={layer.title}
           onDelete={() => onLayerDelete(layer.id)}
           onSelect={() => onLayerSelect(layer.id)}
+          onLayerNameUpdate={onLayerNameUpdate}
         />
       ))}
     </LayerContainer>
   );
 };
 
 type LayerItemProps = {
+  id: string;
   layerTitle: string;
   onDelete: () => void;
   onSelect: () => void;
+  onLayerNameUpdate: (inp: LayerNameUpdateProps) => void;
 };
 
-const LayerItem: React.FC<LayerItemProps> = ({ layerTitle, onDelete, onSelect }) => {
-  const [isMenuOpen, setMenuOpen] = React.useState(false);
+const LayerItem = ({ id, layerTitle, onDelete, onSelect, onLayerNameUpdate }: LayerItemProps) => {
+  const [isMenuOpen, setMenuOpen] = useState(false);
+  const toggleMenu = useCallback(() => {
+    setMenuOpen(prev => !prev);
+  }, []);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(layerTitle);
+  const [prevTitle, setPrevTitle] = useState(layerTitle);
 
-  const toggleMenu = () => setMenuOpen(prev => !prev);
+  const handleDoubleClick = useCallback(() => {
+    setIsEditing(true);
+    setPrevTitle(editedTitle);
+  }, [editedTitle]);
+
+  const handleTitleSubmit = useCallback(() => {
+    if (!editedTitle.trim()) {
+      return;
+    }
+    setIsEditing(false);
+    onLayerNameUpdate({ layerId: id, name: editedTitle });
+  }, [editedTitle, id, onLayerNameUpdate]);
+
+  const handleKeyUp = useCallback(
+    (e: { key: string }) => {
+      if (["Enter", "Return"].includes(e.key)) {
+        handleTitleSubmit();
+      }
+    },
+    [handleTitleSubmit],
+  );
+
+  const handleBlur = () => {
+    setIsEditing(false);
+    setEditedTitle(prevTitle);
+  };
```

Comment:
For other inputs clicking away will set the new value (aka save the value). Can you make handleblur call onLayerNameUpdate? 🙏🏻

Reply:
Also useCallback

Reply:
Done.